build_runner/lib/src/generate/watch_impl.dart

```diff
@@ -117,13 +117,16 @@ class WatchImpl implements BuildState {
   /// Checks if we should skip a watch event for this [change].
   bool _shouldProcess(AssetChange change) {
     if (_isCacheFile(change)) return false;
+    if (_isGitFile(change)) return false;
     if (_isEditOnGeneratedFile(change)) return false;
     if (_isExpectedDelete(change)) return false;
     if (_isUnwatchedDelete(change)) return false;
     return true;
   }
 
-  bool _isCacheFile(AssetChange change) => change.id.path.contains(cacheDir);
+  bool _isCacheFile(AssetChange change) => change.id.path.startsWith(cacheDir);
+
+  bool _isGitFile(AssetChange change) => change.id.path.startsWith('.git/');
```

Comment:
Not a windows-friendly path.

Better to use pkg/path?

Reply:
In theory `AssetId` is supposed to already be normalized for us but we might have bugs. If so I'd rather fix them there than here.

Reply:
https://github.com/dart-lang/build/blob/cca7237378ff45c7858158c07c501072c2ffe1f0/build/lib/src/asset/id.dart#L146-L156